src/global_logic.js

```diff
@@ -786,7 +786,7 @@ module.exports.calcBasedOneSummon = function (summonind, prof, buff, totals) {
             }
         } else {
             // Normal attack * n times
-            var expectedCycleDamage = expectedTurn * expectedAttack * damage;
+            let expectedCycleDamage = expectedTurn * expectedAttack * damage;
```

Comment:
:+1:

If it was in a small function, then does not care var/let. it's a narrow scope.
Now, here it's in long logic function, limit scope by the let keyword make the code strict.

one more change request.
- [ ] declare var expectedCycleDamagePerTurn outside of if-block.

```jsx
var expectedCycleDamagePerTurn;
if (expectedTurn === Infinity) {
    expectedCycleDamagePerTurn = ...
    : 
} else {
    let expectedCycleDamage = ...
    : 
   expectedCycleDamagePerTurn = ...
}
```

I found motocal follow the rule. (but not all, as you know most part uses var)

```jsx
var hp;
if (displayRealHP) {
    // Actual HP
    hp = parseInt(0.01 * (k + 1) * onedata["Djeeta"].totalHP);
} else {
    // Residual HP ratio
    hp = k + 1
}
```

----
https://eslint.org/docs/rules/no-inner-declarations